boa_engine/src/error/mod.rs

```diff
@@ -0,0 +1,767 @@
+    /// ```
+    pub fn error() -> Self {
+        Self::new(JsNativeErrorKind::Error, Box::default(), None)
+    }
+
+    /// Creates a new `JsNativeError` of kind `EvalError`, with empty `message` and undefined `cause`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use boa_engine::{JsNativeError, JsNativeErrorKind};
+    /// let error = JsNativeError::eval();
+    ///
+    /// assert!(matches!(error.kind, JsNativeErrorKind::Eval));
+    /// ```
+    pub fn eval() -> Self {
+        Self::new(JsNativeErrorKind::Eval, Box::default(), None)
+    }
+
+    /// Creates a new `JsNativeError` of kind `RangeError`, with empty `message` and undefined `cause`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use boa_engine::{JsNativeError, JsNativeErrorKind};
+    /// let error = JsNativeError::range();
+    ///
+    /// assert!(matches!(error.kind, JsNativeErrorKind::Range));
+    /// ```
+    pub fn range() -> Self {
+        Self::new(JsNativeErrorKind::Range, Box::default(), None)
+    }
+
+    /// Creates a new `JsNativeError` of kind `ReferenceError`, with empty `message` and undefined `cause`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use boa_engine::{JsNativeError, JsNativeErrorKind};
+    /// let error = JsNativeError::reference();
+    ///
+    /// assert!(matches!(error.kind, JsNativeErrorKind::Reference));
+    /// ```
+    pub fn reference() -> Self {
+        Self::new(JsNativeErrorKind::Reference, Box::default(), None)
+    }
+
+    /// Creates a new `JsNativeError` of kind `SyntaxError`, with empty `message` and undefined `cause`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use boa_engine::{JsNativeError, JsNativeErrorKind};
+    /// let error = JsNativeError::syntax();
+    ///
+    /// assert!(matches!(error.kind, JsNativeErrorKind::Syntax));
+    /// ```
+    pub fn syntax() -> Self {
+        Self::new(JsNativeErrorKind::Syntax, Box::default(), None)
+    }
+
+    /// Creates a new `JsNativeError` of kind `TypeError`, with empty `message` and undefined `cause`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use boa_engine::{JsNativeError, JsNativeErrorKind};
+    /// let error = JsNativeError::typ();
+    ///
+    /// assert!(matches!(error.kind, JsNativeErrorKind::Type));
+    /// ```
+    pub fn typ() -> Self {
+        Self::new(JsNativeErrorKind::Type, Box::default(), None)
+    }
+
+    /// Creates a new `JsNativeError` of kind `UriError`, with empty `message` and undefined `cause`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use boa_engine::{JsNativeError, JsNativeErrorKind};
+    /// let error = JsNativeError::uri();
+    ///
+    /// assert!(matches!(error.kind, JsNativeErrorKind::Uri));
+    /// ```
+    pub fn uri() -> Self {
+        Self::new(JsNativeErrorKind::Uri, Box::default(), None)
+    }
+
+    /// Appends a message to this error.
```

Comment:
```suggestion
    /// Sets the message of this error.
```
Don't like using the word append here, it gives the idea of extending a String of Vec.
Same for `with_cause` below